### PR TITLE
SONARIAC-410 [K8] Rule S6429: Exposing Docker sockets is security-sensitive

### PR DESCRIPTION
**Pull request by marco-bearzi-sonarsource**

### CODE REVIEW

File: iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java [hunk 1]
```diff
@@ -32,6 +32,11 @@ void isTrue() {
     assertThat(TreePredicates.isTrue().test(text("true"))).isTrue();
   }
 
+  @Test
+  void isEqualTo() {
+    assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text("VALUE_TEST"))).isTrue();
```
Comment (nils-werner-sonarsource) [inline iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java:37]: Maybe you add another negative test, as well as a test where none TextTree is tested in the predicate. (Yes, I know the other predicate is missing this test as well. If possible, you might add it there as well)

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DockerSocketCheck.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import java.util.List;
+import org.sonar.check.Rule;
+
+import static org.sonar.iac.common.yaml.TreePredicates.isEqualTo;
+
+@Rule(key = "S6429")
+public class DockerSocketCheck extends KubernetesObjectCheck {
+
+  private static final String MESSAGE = "Make sure exposing the Docker socket is safe here.";
+  @SuppressWarnings("java:S1075")
```
Comment (nils-werner-sonarsource) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DockerSocketCheck.java:31]: Good job! Maybe you can add a small explanation.

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! See small suggestions.

File: iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java [hunk 3]
```diff
@@ -30,11 +30,15 @@ class TreePredicatesTest {
...
     assertThat(TreePredicates.isTrue().test(text("true"))).isTrue();
+    assertThat(TreePredicates.isTrue().test(text("false"))).isFalse();
+    assertThat(TreePredicates.isTrue().test(text(null))).isFalse();
   }
 
   @Test
   void isEqualTo() {
     assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text("VALUE_TEST"))).isTrue();
+    assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text("NOT_VALUE_TEST"))).isFalse();
+    assertThat(TreePredicates.isEqualTo("VALUE_TEST").test(text(null))).isFalse();
```
Comment (nils-werner-sonarsource) [inline iac-common/src/test/java/org/sonar/iac/common/yaml/TreePredicatesTest.java:41]: I was more thinking about another kind of tree which is not a ScalarTree aka TextTree

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM. Please check comment
